docs-src/app-dev-context/cancel-activity.md

```diff
@@ -0,0 +1,18 @@
+---
+id: cancel-activity
+title: Cancel an Activity from a Workflow
+description: If an Activity is supposed to react to Cancellation.
+sidebar_label: Cancel an Activity
+tags:
+  - guide-context
+---
+
+Cancelling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout. If the Heartbeat is not invoked, the Activity cannot receive a cancellation request. Users should set Heartbeat and set a [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout) when executing any non-immediate Activity to ensure the server knows it is still working.
+
+When an Activity is cancelled, an error is raised in the Activity at the next available opportunity. Temporal recommends catching the error and perform clean-up logic to ensure that the task is properly completed.
```

Comment:
```suggestion
When an Activity is cancelled, an error is raised in the Activity at the next available opportunity. If cleanup logic needs to be performed, it can be done in a finally or inside a caught cancel error, but in order for the activity to appear cancelled the exception needs to be re-raised.
```

We definitely never want recommending catching the error unless they absolutely have to.